docs/api/menu-item.md

```diff
@@ -14,7 +14,7 @@ See [`Menu`](menu.md) for examples.
     * `menuItem` MenuItem
     * `window` [BaseWindow](base-window.md) | undefined - This will not be defined if no window is open.
     * `event` [KeyboardEvent](structures/keyboard-event.md)
-  * `role` string (optional) - Can be `undo`, `redo`, `cut`, `copy`, `paste`, `pasteAndMatchStyle`, `delete`, `selectAll`, `reload`, `forceReload`, `toggleDevTools`, `resetZoom`, `zoomIn`, `zoomOut`, `toggleSpellChecker`, `togglefullscreen`, `window`, `minimize`, `close`, `help`, `about`, `services`, `hide`, `hideOthers`, `unhide`, `quit`, `showSubstitutions`, `toggleSmartQuotes`, `toggleSmartDashes`, `toggleTextReplacement`, `startSpeaking`, `stopSpeaking`, `zoom`, `front`, `appMenu`, `fileMenu`, `editMenu`, `viewMenu`, `shareMenu`, `recentDocuments`, `toggleTabBar`, `selectNextTab`, `selectPreviousTab`, `showAllTabs`, `mergeAllWindows`, `clearRecentDocuments`, `moveTabToNewWindow` or `windowMenu` - Define the action of the menu item, when specified the
+  * `role` string (optional) - Can be `undo`, `redo`, `cut`, `copy`, `paste`, `pasteAndMatchStyle`, `delete`, `selectAll`, `reload`, `forceReload`, `toggleDevTools`, `resetZoom`, `zoomIn`, `zoomOut`, `toggleSpellChecker`, `togglefullscreen`, `window`, `minimize`, `close`, `help`, `about`, `services`, `hide`, `hideOthers`, `unhide`, `quit`, `showSubstitutions`, `toggleSmartQuotes`, `toggleSmartDashes`, `toggleTextReplacement`, `startSpeaking`, `stopSpeaking`, `zoom`, `front`, `appMenu`, `fileMenu`, `editMenu`, `viewMenu`, `shareMenu`, `recentDocuments`, `toggleTabBar`, `selectNextTab`, `selectPreviousTab`, `showAllTabs`, `mergeAllWindows`, `clearRecentDocuments`, `moveTabToNewWindow`, `windowMenu`, `palette` or `header` - Define the action of the menu item, when specified the
     `click` property will be ignored. See [roles](#roles).
```

Comment:
Is this still true for the `palette` role? It's not super clear without a full example, but it kinda sounds like the `palette` role would specify how the menu item is rendered but not necessarily change how the click property is handled.

Reply:
The fact that `palette` is an `NSMenuPresentationStyle` property rather than a role makes me wonder if we need a different property for this?

Reply:
I went with `role`, because that seemed to most fitting for this extra functionality. I was wondering about `type` but that's not the best either since we also need to pass a `submenu` for the `palette` layout to work. Also didn't want to introduce a completely new property since this is a quite specific macOS-only feature.
Do you have any suggestion for a better place for it?

Reply:
I think based on the most recent API meeting, we might be confused about the `role` API itself. Many of the roles involve specific actions to take (copy, paste), but it's actually a mixed bag of various native behaviors.

It seems like overtime the API has been poorly reused, imo. `role` sometimes serves as a predefined template while others it completely overrides the native behavior.

We'll discuss this once more this Thursday for hopefully final approval.

Reply:
Can we use `type` for the header similar to how we implement `separator`?
https://github.com/electron/electron/blob/d8baceb08caae374d3931d5747152e3971a1a379/shell/browser/ui/cocoa/electron_menu_controller.mm#L210-L212

Reply:
Similarly, we were thinking `palette` could fit under the `type` property as well. Given that it's a kind of subtype of `submenu`.